lib/std/child_process.zig

```diff
@@ -156,6 +156,7 @@ pub const ChildProcess = struct {
         Ignore,
         Pipe,
         Close,
+        File, // Pass an open file to the process. This will take ownership of the file, so **do not close it.**
```

Comment:
> This will take ownership of the file, so **do not close it.**

1. This is too vague. Of course the file can be closed in the parent after spawning the child.
2. There is no test for this behavior.
3. Afaiu, the file descriptor may still leak after dup is called to parallel spawned processes before it is closed, since on the duped file descriptor the CLOEXEC is not set. On Windows https://learn.microsoft.com/en-us/windows/win32/sysinfo/handle-inheritance is needed for the file to be inheritable. This is missing as comments.

Reply:
(My apologies for taking so long to respond to this) Yeah, I realize that now, so it doesn't really need to be there. Regarding a test, where should I put it? Would the bottom of `std/child_process.zig` work? I think a good one would be to create a new file, use `echo` with its stdout set to the file, and test if the output is as expected, but if that doesn't seem thorough enough let me know and I'll try to think of something else (or take your suggestion, whatever you want). 

As far as I can see, Windows doesn't have an equivalent of CLOEXEC. Its docs say that to ensure you don't leak resources, you should be sure to close the handles in the child and the parent. The file lacking the inheritance bit at first isn't required, because in the `dupWinHandleInherit` function I clone it and mark it inheritable but with the same permissions. The idea of not closing the file in the host is probably a good one to keep, though, I argue. It probably isn't worth it to open a file, have it closed when passed to the child, and then open it again. We could do it this way, though, if that is better for performance or so forth. I will likely adjust the comment to "After spawning the child, you can close this or read from it, depending on your uses," or something of the sort.

Reply:
> Regarding a test, where should I put it?

1. Take a look at test/standalone.zig child_process.zig or windows_spawn.zig
2. No need to use a platform command as dependency. Just read the file content from the child process and compare a static fixed test, which is identical to the parent process. Make sure to create a tmpDir and provide the child process with the open file descriptor to the file inside the tmpDir.

> Windows doesn't have an equivalent of CLOEXEC

Windows mandates that you first enable inheritance in the spawned process and only then it inherits the things you provide. You can further restrict it via providing a security handle with attributes for more cases and finally, you can provide it a meta structure, which sounds like it closes in the kernel via callback etc the unmarked handles.

1. As far as I understand, that is not true at least for this case. See for the discussion and implementation in the linked issue, where /dev/null was provided. If you think otherwise, then please argue why the same approach would not work. You can use the security attributes to only provide stdin, stdout, stderr as shown in the linked issue.

Personally I think of security attributes and meta structures as poorly solved callback for a proper in-Kernel VM-based "do these things to not leak ressources" solution, but at least Linux looks like it is getting there eventually and the others will catch up eventually for performance and security.

Reply:
1. Neato, I shall put the test there. I mentioned echo because that's a fairly universal command that I could use as the child process, but if that's bad I'll think of something else.

2. I don't think it could be done via security attributes in the same way as #15565 because `windowsCreateProcessPathExt` doesn't accept a security attributes pointer. In the aforementioned issue, he marks the Null handle as writable and readable. He removes the need for setting the handle information by adding the ability to mark files opened through `windows.OpenFile` as inheritable based on the value of `SECURITY_ATTRIBUTES.bInheritHandle`. I can't do that in the same way because I already have an open handle, which I instead mark as inheritable through the `dupWinHandleInherit` function. If I am misunderstanding something, please forgive me.

Reply:
> I can't do that in the same way because I already have an open handle, which I instead mark as inheritable through the dupWinHandleInherit function. If I am misunderstanding something, please forgive me.

Why does for example https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-reopenfile not work?